tests/tests/tx.rs

```diff
@@ -190,6 +198,142 @@ async fn dry_run_above_block_gas_limit() {
+            let tx =
+                arb_large_script_tx(189028 + i as Word, script_bytes_count, &mut rng);
+            let size = tx
+                .as_script()
+                .expect("script tx expected")
+                .metered_bytes_size() as u64;
+            (tx, size)
+        })
+        .collect();
+
+    // Run 5 cases. Each one will allow one more transaction to be included due to size.
+    for n in 1..=arb_tx_count {
+        // Calculate proper size limit for 'n' transactions
+        let block_transaction_size_limit: u64 =
+            transactions.iter().take(n).map(|(_, size)| size).sum();
+        let config = config_with_size_limit(block_transaction_size_limit);
+        let srv = FuelService::new_node(config).await.unwrap();
+        let client = FuelClient::from(srv.bound_address);
+
+        // Submit all transactions
+        for (tx, _) in &transactions {
+            let status = client.submit(tx).await;
+            assert!(status.is_ok())
+        }
+
+        let _ = tokio::time::sleep(Duration::from_millis(10)).await;
+
+        // Produce a block.
+        let block_height = client.produce_blocks(1, None).await.unwrap();
+
+        // Assert that only 'n' first transactions (+1 for mint) were included.
+        let block = client.block_by_height(block_height).await.unwrap().unwrap();
+        assert_eq!(block.transactions.len(), n + 1);
+        let expected_ids: Vec<_> = transactions
+            .iter()
+            .take(n)
+            .map(|(tx, _)| tx.id(&ChainId::default()))
+            .collect();
+        let actual_ids: Vec<_> = block.transactions.into_iter().take(n).collect();
+        assert_eq!(expected_ids, actual_ids);
+    }
+}
+
+#[tokio::test]
+async fn transaction_selector_can_select_a_transaction_that_fits_the_block_size_limit() {
+    let mut rng = rand::rngs::StdRng::from_entropy();
+
+    // Create 5 transactions of decreasing sizes.
+    let arb_tx_count = 5;
+    let transactions: Vec<(_, _)> = (0..arb_tx_count)
+        .map(|i| {
+            let script_bytes_count = 10_000 + (i * 100);
+            let tx =
+                arb_large_script_tx(189028 + i as Word, script_bytes_count, &mut rng);
+            let size = tx
+                .as_script()
+                .expect("script tx expected")
+                .metered_bytes_size() as u64;
+            (tx, size)
+        })
+        .rev()
+        .collect();
+
+    let (smallest_tx, smallest_size) = transactions.last().unwrap();
+
+    // Allow only the smallest transaction to fit.
+    let config = config_with_size_limit(*smallest_size);
+    let srv = FuelService::new_node(config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    // Submit all transactions
+    for (tx, _) in &transactions {
+        let status = client.submit(tx).await;
+        assert!(status.is_ok())
+    }
+
+    let _ = tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
The same here, do we need to wait if you use manual block production?

Reply:
See: https://github.com/FuelLabs/fuel-core/pull/2195#discussion_r1769589697